src/main.rs

```diff
@@ -672,7 +688,26 @@ fn exec_cargo_inner(
     if progress.count != 0 && !cx.print_command_list && cx.log_group == LogGroup::None {
         eprintln!();
     }
-    progress.count += 1;
+
+    let new_count = progress.count + 1;
+    let mut skip = false;
+    if let Some(partition) = &cx.partition {
+        if progress.count % partition.count != partition.index - 1 {
+            let mut msg = String::new();
```

Comment:
IIUC this would be something like:
```
1/4 run
2/4 skipped
3/4 run
4/4 skipped
```


I think this is fine for the nextest use, I guess it maybe a bit inefficient for cargo-hack use, since the feature combinations that would reduce the amount of recompilation are often before or after the current combination.

Reply:
thanks for nice suggestion. I've switched from interleaved partitioning to chunked partitioning. and seems our ci got quicker to finish: https://buildkite.com/anza/agave/builds/7952#0190d3f9-52af-4b75-9e69-e408f4a1a57e